FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download_Multithreaded/DemoTasks/S3DownloadMultithreadedHTTPExample.c

```diff
@@ -0,0 +1,1142 @@
+/*
+ * FreeRTOS V202011.00
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * https://www.FreeRTOS.org
+ * https://github.com/FreeRTOS
+ *
+ */
+
+/*
+ * Demo for showing the use of the HTTP API using a server-authenticated network
+ * connection.
+ *
+ * This example resolves a S3 domain (using a pre-signed URL), establishes a TCP
+ * connection, validates the server's certificate using the configurable root CA
+ * certificate, and then finally performs a TLS handshake with the HTTP server,
+ * so that all communication is encrypted.
+ *
+ * Afterwards, a request and a response task are created, in addition to the
+ * main demo task, along with two thread-safe queues (a request and response
+ * queue), to demonstrate an asynchronous workflow for downloading an S3 file.
+ *
+ * The main task is responsible for sending HTTP requests to the server over the
+ * transport interface, using the HTTP Client library API. It reads requests
+ * from the request queue and places the corresponding server response on the
+ * response queue. The request task is responsible for generating these
+ * requests, specifying a byte range with each iteration in order to download
+ * the S3 file in partial content responses. Finally, the response task logs and
+ * evaluates the responses to these requests. The tasks run continuously until
+ * the entire file is downloaded. If any request fails, an error code is
+ * returned.
```

Comment:
This style of description for different tasks involved will be improving the readability. https://github.com/FreeRTOS/FreeRTOS/blob/master/FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Multitask/DemoTasks/MultitaskMQTTExample.c#L46.